src/type/dto/got-type.dto.ts

```diff
@@ -9,9 +9,15 @@ export class GotTypeDto {
     @IsNotEmpty()
     readonly name: string;
 
+    @ApiModelProperty({ type: Object, isArray: true })
+    readonly validators: any[];
+
     @ApiModelProperty({ type: GotPropertyDto })
     @ValidateNested()
     @Type(() => GotPropertyDto)
     readonly properties: GotPropertyDto[];
 
+    @ApiModelProperty({ type: Boolean })
+    readonly required: boolean;
```

Comment:
Why should a type be required or not?

Reply:
In practice following scenarios will occur:
A) You want to define an object as mandatory, even though not all of it's properties are required.
B) You want to define an object as NOT required, but still some or all of it's properties as required. This could happen, when you want to ensure that the user should either provide no object or if he decides to provide the object, he obviously is forced to provide the mandatory sub properties

Reply:
Therefore having the option to set a required flag not only for properties but also for complex types is something very common and already done similar in popular decscription languages like swagger